gzwrite.c

```diff
@@ -335,8 +334,12 @@ int ZEXPORT PREFIX(gzputs)(gzFile file, const char *s) {
 
     /* write string */
     len = strlen(s);
-    ret = (int)gz_write(state, s, len);
-    return ret == 0 && len != 0 ? -1 : ret;
+    if ((int)len < 0 || (unsigned)len != len) {
```

Comment:
Easier than cast is to just compare ```len > INT_MAX``` as that macro should be defined in all modern compilers.